### ISSUE TITLE
[Request] Option to wrap if line break exists

### ISSUE THREAD

**Issue by jrj2211** [electrovir/prettier-plugin-multiline-arrays#23]
The one thing I've seen requested time and time again for prettier is to support formatting arrays the same as objects. I'm curious what the feasibility of that is and if it's in scope for this plugin as it deals with multi line arrays.

I have a hard time wanting to put a hard controlled number for when I want my arrays to wrap. There's just to many scenarios where sometimes I want it to wrap and other times I don't. I think that's why this topic has been so heated and heavily discussed in their issues.

For example, A small array like this is totally readable and I'd want it on 1 line. But the 2nd example is much uglier as a single line.
```
const methods = ['find', 'get', 'create', 'patch', 'remove'];
const hooks = [authenticate('jwt'), schemaHooks.resolveExternal(timeShiftsExternalResolver), schemaHooks.resolveResult(timeShiftsResolver)];
```

When objects are formatted, its based off if the first element is on a line break. For example, this would get merged into a single line (unless its over the line length):

```
// Input
const test = { method: 'cash',
  status: 'paid',
};

// Formatted
const test = { method: 'cash', status: 'paid' };
```

But, if the first item uses a line break, then the rest of the array is expanded:
```
// Input
const test = {
  method: 'cash', status: 'paid',
};

// Formatted
const test = {
  method: 'cash', 
  status: 'paid',
};
```

I'd really love to see this for arrays as it lets you decide when programming what formatting makes the most sense, instead of being one or the other with no flexibility based on the situation. Is this within scope of this plugin?

**Comment by jrj2211** [comment]
Awesome, thank you!! This is my only frustration with prettier and I've contemplated going back to just eslint numerous times over the last year because of it haha. 

If you accept donations I'd love to send some your way for the help!

**Comment by electrovir** [comment]
You can actually use Prettier and ESLint together @jrj2211, though I admit my own current dev pipeline doesn't do that, hah.

I've setup a [GitHub sponsors profile](https://github.com/sponsors/electrovir) if you're into that.

**Comment by jrj2211** [comment]
Sorry, didn't see your last comment until just now. Sent a little your way through that link, hopefully it'll buy a couple lunches.

**Comment by electrovir** [comment]
Thanks! ❤️

Sorry for the delay on this one. I got the first step working already but I'm still debating how to handle this fighting with the wrap threshold. Perhaps you have some input on this?

Some possible ideas I have:
1. new Prettier config option to disable the wrap threshold, only force Prettier to wrap on the manual conditions (trailing comma, leading newline)
2. by default do not apply a wrap threshold and only force wrapping on the manual wrap conditions (leading newline, trailing comma); Prettier will still wrap based on line length. To apply a wrap threshold, `multilineArraysWrapThreshold` must be provided to Prettier's config.
3. Just leave it the way it is (default wrap threshold of 1).

**Comment by electrovir** [comment]
I'm leaning towards 2. That would possibly disrupt current users, but a major version bump (to 2.0.0) should be enough to indicate such a change 🤔

**Comment by jrj2211** [comment]
No need to apologize, you're doing me a favor. I'm not in any rush either.

Hmm... If you're okay with changing the default behavior of the plugin then I think that makes sense. Essentially that's making this mode the new default?

If you wanted to keep backwards compatibility, instead of a new option I'd also be okay with setting 'multilineArraysWrapThreshold` to 0 to turn it off instead of a new option if that already doesn't do something else.

**Comment by Shakeskeyboarde** [comment]
+1 to setting `multilineArraysWrapThreshold` to 0.

**Comment by electrovir** [comment]
Okay I finally finished #24 so I'm back on this. Everything is working for detecting leading new lines, I just need to modify how options are figured out now.

> +1 to setting `multilineArraysWrapThreshold` to 0.

The issue with this options is that it prevents anyone from setting that option to `0` intentionally (if someone want's _all_ arrays to _always_ wrap). This is why I think option 2 from above is the best:

> 2\. by default do not apply a wrap threshold and only force wrapping on the manual wrap conditions (leading newline, trailing comma); Prettier will still wrap based on line length. To apply a wrap threshold, `multilineArraysWrapThreshold` must be provided to Prettier's config.

I'll include a major version bump and a detailed changelog explaining the change.

**Comment by electrovir** [comment]
Just published this in [version 2.0.0](https://www.npmjs.com/package/prettier-plugin-multiline-arrays/v/2.0.0). Going to try it out in my larger projects before closing this ticket.

**Comment by electrovir** [comment]
Everything is looking good so far, so I'm closing this now! 🥳
